runtime/bcverify/rtverify.c

```diff
@@ -1531,15 +1531,26 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 				 * cases of invokevirtual which only invoke public methods of a public class.
 				 */
 
+				/* Remove the receiver from the stack. */
+				type = POP;
 				/* Receiver compatible with MethodHandle? */
-				type = POP;		/* Remove the receiver from the stack */
-				rc = isClassCompatibleByName (verifyData, type, (U_8 *)"java/lang/invoke/MethodHandle", sizeof("java/lang/invoke/MethodHandle") - 1, &reasonCode);
+				rc = isClassCompatibleByName(verifyData, type, (U_8 *)"java/lang/invoke/MethodHandle", sizeof("java/lang/invoke/MethodHandle") - 1, &reasonCode);
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+				if (FALSE == rc) {
+					if (BCV_ERR_INSUFFICIENT_MEMORY == reasonCode) {
+						goto _outOfMemoryError;
+					}
+					/* Receiver compatible with VarHandle? */
+					rc = isClassCompatibleByName(verifyData, type, (U_8 *)"java/lang/invoke/VarHandle", sizeof("java/lang/invoke/VarHandle") - 1, &reasonCode);
```

Comment:
is this applicable to both the `JBinvokehandle` and the  `JBinvokehandlegeneric` cases?

Reply:
Just the `invokehandle` scenario. I should skip the `VarHandle` check for `JBinvokehandlegeneric`.